app/models/vaccination_record.rb

```diff
@@ -185,6 +186,8 @@ class VaccinationRecord < ApplicationRecord
               less_than_or_equal_to: -> { Time.current }
             }
 
+  validate :nhs_immunisations_api_primary_source_consistency
```

Comment:
```suggestion
  validates :nhs_immunisations_api_primary_source,
            inclusion: [true, false],
            if: :nhs_immunisations_api_id
```

This would be a more Rails way of doing it and then the string can be put in the locale file.